src/mumble/UserModel.cpp

```diff
@@ -1271,6 +1271,10 @@ void UserModel::setComment(Channel *c, const QString &comment) {
 					QToolTip::showText(QCursor::pos(), data(index(c, 0), Qt::ToolTipRole).toString(),
 									   Global::get().mw->qtvUsers);
 				}
+			} else if (c->iId == static_cast< unsigned int >(~iChannelDescription)) {
+				iChannelDescription               = -1;
```

Comment:
The whole casting here looks somewhat fishy. It is probably working as expected (and I realize you replicated the same mechanism as for the user comment).
I _think_ this might theoretically break when reaching like 2 billion channels, which would be fine. But have considered edge cases here? ``root`` has channel id ``0`` which would be ``~0 == -1`` when considering signed integers.

*time passes*

Haha, ok guess what. This indeed breaks for ``root``. I just tried it. If you set the description of the root channel, a message will popup every time the content is received from the server. (e.g. when reconnecting).
Sadly as of posting this I have no solid idea how to fix this.

Reply:
The best thing I can think of without refactoring receiving descriptions/comments from the server is to add an extra field to `Channel` and `ClientUser` which is an enum indicating what the request is for (popup vs. "view description" with the flexibility to support more cases in the future)

If this is acceptable, I can try implementing it soon

Reply:
the problem doesn't exist in ClientUser since users never get session id 0, but using the negation of the IDs to differentiate between popup and the "view description/comment" option is hacky in both cases

Reply:
The extra field as an `enum` sounds good to me.

Reply:
I am all for switching to a "proper" implementation instead of negating ids. I do not think anything will break if you remove the existing negation logic.